stlab/channel.hpp

```diff
@@ -348,6 +359,7 @@ struct shared_process : shared_process_receiver<yield_type<T, Arg>>,
                 cts = _process_message_queue.size() == 0; // TODO (sparent) : queue size - 1.
```

Comment:
This line is assuming the process_queue_size is 1. Replace the "0" with _process_queue_size - 1.